### PR TITLE
Minor: Clean up user types

### PR DESCRIPTION
**Pull request by BillyZac**
Addresses [the list users story](https://kanbanflow.com/t/096fa42b062dbc8c82febb2a0c3f3707/3-as-an-admin-i-want-to-list-users-m).

- Adds explicit types to the user service methods.

- In an attempt to keep the user service nice and "declarative", moves a few utilities into a `user_functions` file.

- We eventually want all models to be kept in the `models` folder. This
commit moves us in that direction, at least a little bit.

### CODE REVIEW

File: src/services/users/user_functions.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import {MSUser} from '../../model/MSUser';
+import {MSContact} from '../../model/MSContact';
+import {BookitUser} from '../../model/BookitUser';
+
```
Comment (mathomas) [inline src/services/users/user_functions.ts]: Do these maybe deserve tests?

Comment (BillyZac) [inline src/services/users/user_functions.ts]: @mathomas I'm not sure what you mean. Are you saying we should test the models? What would that look like?

Comment (mathomas) [inline src/services/users/user_functions.ts]: No, the functions (mapMSUserToBookitUser, mapMSContactToBookitUser) that map from one type to another.  Sorry it wasn't clear.  Didn't know where else to drop the comment other than directly over the functions.  I looked for tests, but didn't see any.  Maybe I missed them.

**Comment by BillyZac** [comment]
@mathomas Good call. I'll add tests and make another PR. Closing for now.

**Pull request closed without merging**